### PR TITLE
Update `dependabot.yml` to make Dependabot update GitHub actions

### LINKED ISSUE #1046: Make dependabot update github actions
## Problem

Some GH action dependencies are behind the latest updates:
- actions/setup-python in CI + CD / Lint job
- docker/setup-buildx-action in CI + CD / Build images (ingestion_server) job
- etc.

## Description

Let's have **dependabot** keep them up to date. The catalog is already configured to do this:

https://github.com/WordPress/openverse-catalog/blob/5933f712d2d017eb1d952c68fffc6f1606d58eb1/.github/dependabot.yml#L25-L33

Whoever takes this can very well copy those lines, and it should work.

## Alternatives

Maybe using [Renovate](https://github.com/renovatebot/renovate), but dependabot is already serving us well. I don't see the need to change it.

## Additional context

- [Configuration options for the dependabot.yml file](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file)
- See an example of warnings thrown by old GH actions in https://github.com/WordPress/openverse-api/actions/runs/3696820240

### PR DESCRIPTION
**Pull request by zmwaris1**
## Fixes

Fixes #1046  by @krysal 

## Description

Some GH action dependencies are behind the latest updates:

+ actions/setup-python in CI + CD / Lint job
+ docker/setup-buildx-action in CI + CD / Build images (ingestion_server) job
+ etc.
Made changes in **dependabot.yml** so that GH dependencies are automatically updated.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zmwaris1** [comment]
As stated in the issue I have made the changes in the required file, if any issues persists please reach out, I will take a look as soon as possible.

File: .github/dependabot.yml
Comment (dhruvkb) [inline .github/dependabot.yml:37]: To keep the file same as the [one in the catalog](https://github.com/WordPress/openverse-catalog/blob/main/.github/dependabot.yml).

```suggestion
  - package-ecosystem: "github-actions"
```

File: .github/dependabot.yml
Comment (dhruvkb) [inline .github/dependabot.yml:40]:
```suggestion
      interval: "weekly"
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]
The changes look good! We'll update the older actions once this PR is merged and Dependabot creates PRs for them.

File: .github/dependabot.yml
Comment (dhruvkb) [inline .github/dependabot.yml:36]: Remove the spaces here to make the lint checks pass.

```suggestion

```

**Review by dhruvkb — CHANGES_REQUESTED** [review]
The lint is failing because of spaces on a blank line.

**Comment by zmwaris1** [comment]
@dhruvkb thanks for the reply, I made the changes as proposed, please have a look.

**Review by sarayourfriend — APPROVED** [review]
LGTM. Thanks for the contribution, @zmwaris1. Just a heads up that our PRs require two approvals to merge and most of the Openverse maintainers are busy this week with offline work, so it might not be until next week for someone else to review this. Additionally, please avoid removing the automatically assigned reviewers requested for PRs in the future as it can cause the PR to fall off the radars of the assigned reviewers and delay reviews.

**Comment by sarayourfriend** [comment]
No worries! Thanks again for the contribution, we do appreciate it, and welcome to Openverse :slightly_smiling_face:

File: .github/dependabot.yml
Comment (dhruvkb) [inline .github/dependabot.yml:35]:
```suggestion
      - dependencies

```

**Review by dhruvkb — CHANGES_REQUESTED** [review]
This is good to merge but the lack of a line separation is bugging me. I tried pushing a commit to this PR but was denied (presumably because it's a PR from `main`) so @zmwaris1, if you could make the change, I'll merge the PR right away.

**Comment by zmwaris1** [comment]
@dhruvkb I made the changes as proposed thanks for the help.

**Pull request merged**
